lib/exercises.rb

```diff
@@ -1,19 +1,40 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  words = Hash.new
+
+  strings.each do |string|
+    letter_count = Hash.new
+    string.each_char do |c|
+      letter_count[c] = letter_count[c] ? letter_count[c] + 1 : 1
+    end
+
+    if words[letter_count]
+      words[letter_count].push(string)
+    else
+      words[letter_count] = [string]
+    end
+  end
+
+  return words.values
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
 # Time Complexity: ?
 # Space Complexity: ?
 def top_k_frequent_elements(list, k)
```

Comment:
This *mostly* works, but it doesn't pass all the tests because it doesn't select the 1st element which appears the most times, in case of a tie.